diff --git a/src/Wt/WDate.C b/src/Wt/WDate.C
index b58b7bd..c85a24b 100644
--- a/src/Wt/WDate.C
+++ b/src/Wt/WDate.C
@@ -624,11 +624,11 @@ bool WDate::writeSpecial(const std::string& f, unsigned&i,
 	if (f[i + 3] == 'M') {
 	  // 4 M's
 	  i += 3;
-	  result << longMonthName(month_).toUTF8();
+	  result << longMonthName(month_, localized).toUTF8();
 	} else {
 	  // 3 M's
 	  i += 2;
-	  result << shortMonthName(month_).toUTF8();
+	  result << shortMonthName(month_, localized).toUTF8();
 	}
       } else {
 	// 2 M's
